simde/simde-math.h

```diff
@@ -235,6 +235,20 @@ SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
   #endif
 #endif
 
+#if !defined(SIMDE_MATH_FLT_MAX)
+  #if defined(FLT_MAX)
+    #define SIMDE_MATH_FLT_MAX FLT_MAX
+  #elif defined(__FLT_MAX__)
+    #define SIMDE_MATH_FLT_MAX __FLT_MAX__
+  #elif defined(__cplusplus)
+    #include <cfloat>
+    #define SIMDE_MATH_FLT_MAX FLT_MAX
+  #else
+    #include <float.h>
+    #define SIMDE_MATH_FLT_MAX FLT_MAX
+  #endif
+#endif
+
```

Comment:
I like that you're doing this instead of just using FLT_MAX/DBL_MAX directly, but I think something like this would be cleaner:

```c
#if !defined(SIMDE_MATH_FLT_MAX)
  #if defined(__FLT_MAX__)
    #define SIMDE_MATH_FLT_MAX __FLT_MAX__
  #else
    #if !defined(FLT_MAX)
      #if defined(__cplusplus)
        #include <cfloat>
      #else
        #include <float.h>
      #endif
    #endif
    #define SIMDE_MATH_FLT_MAX FLT_MAX
  #endif
#endif
```

It may be slightly longer, but it keeps the FLT_MAX case together.

Maybe it's just me, though?  I know some people like to limit nested logic as much as possible…